### PR TITLE
Describe Problems API in release notes

### PR DESCRIPTION
**Pull request by donat**
Fixes https://github.com/gradle/gradle/issues/27349.

### CODE REVIEW

File: platforms/documentation/docs/src/docs/release/notes.md [hunk 1]
```diff
@@ -63,6 +63,18 @@ A new [`named(Spec<String>)` method](javadoc/org/gradle/api/NamedDomainObjectCol
...
+
+Gradle now has a new incubating API that allows for reporting and consuming problems occurring during a build.
+
+The [`Problems`](javadoc/org/gradle/api/problems/Problems.html) service can be used to describe and emit problems with details (description, location information, link to documentation etc).
+Reported problems are then exposed via the Tooling API, allowing Gradle IDE providers - for example, IntelliJ IDEA, Visual Studio Code, Eclipse - to integrate problem details in the UI.
+The reported problems carry location information therefore IDEs can easily integrate them into the developer experience, providing error markers, problem view, and more.
+
+Gradle already emits problems from many components, including (but not limited to) deprecation warnings, dependency version catalog errors, task validation errors, and Java toolchain problems.
+
+The current release focuses on the IDE, however users can expect problem reports appearing in the console in future releases.
```
Comment (lkasso) [inline platforms/documentation/docs/src/docs/release/notes.md:76]:
```suggestion
The current release focuses on reporting problems in the IDE. 
Users can expect further enhancements to the Problems API aimed at console reporting in future releases.
```

**Review by lkasso — APPROVED** [review]
Cleaned up some of the language.

**Comment by donat** [comment]
Thank you @lkasso for the review!

**Comment by donat** [comment]
@bot-gradle merge

**Pull request merged**